src/main/java/fmartin1/model/pokemon/PokemonComparator.java

```diff
@@ -0,0 +1,69 @@
+package fmartin1.model.pokemon;
+
+
+import fmartin1.model.pokemon.type.Type;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+public class PokemonComparator implements Comparator<Pokemon> {
+    private Criteria _sortCriteria;
+
+    public PokemonComparator(Criteria sortCriteria) {
+        super();
+        _sortCriteria = sortCriteria;
+    }
+
+    @Override
+    public int compare(Pokemon o1, Pokemon o2) {
```

Comment:
I would prefer to create different `Comparators` instead of having a big method that selects based on an input. This seems "cochi" to me

Reply:
https://stackoverflow.com/questions/2839137/how-to-use-comparator-in-java-to-sort

Reply:
Split this method to have different comparators